swarm/tools/runs_gc.py

```
@@ -75,12 +76,17 @@ def get_dir_size(path: Path) -> int:
-                    total += entry.stat().st_size
-                except OSError:
-                    pass
+        # Use os.scandir instead of Path.rglob to avoid object creation and stat() overhead
+        # for a significant performance optimization when scanning large directories
+        with os.scandir(path) as it:
+            for entry in it:
+                if entry.is_file():
+                    try:
+                        total += entry.stat().st_size
+                    except OSError:
+                        pass
```

Ruff (0.15.7)

[warning] Use `contextlib.suppress(OSError)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(OSError): ...`

(SIM105)